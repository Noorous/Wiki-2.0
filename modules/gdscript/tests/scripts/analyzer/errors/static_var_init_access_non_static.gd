# GH-83468, GH-91403

@static_unload

func non_static_func():
	return 0

static var static_var_11 = non_static_func()

static var static_var_12:
	set(_value):
		var f := func ():
			var g := func ():
				print(non_static_func())
			g.call()
		f.call()

static var static_var_13 = func ():
	var f := func ():
		var g := func ():
			print(non_static_func())
		g.call()
	f.call()

static var static_var_21a = non_static_func
static var static_var_21b = Callable(non_static_func)

static var static_var_22:
	set(_value):
		var f := func ():
			var g := func ():
				print(non_static_func)
			g.call()
		f.call()

static var static_var_23 = func ():
	var f := func ():
		var g := func ():
			print(non_static_func)
		g.call()
	f.call()

func test():
	pass
