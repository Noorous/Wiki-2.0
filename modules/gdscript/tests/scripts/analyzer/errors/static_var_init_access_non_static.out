GDTEST_ANALYZER_ERROR
>> ERROR at line 8: Cannot call non-static function "non_static_func()" from a static variable initializer.
>> ERROR at line 14: Cannot call non-static function "non_static_func()" from the static function "@static_var_12_setter()".
>> ERROR at line 21: Cannot call non-static function "non_static_func()" from a static variable initializer.
>> ERROR at line 25: Cannot access non-static function "non_static_func" from a static variable initializer.
>> ERROR at line 26: Cannot access non-static function "non_static_func" from a static variable initializer.
>> ERROR at line 32: Cannot access non-static function "non_static_func" from the static function "@static_var_22_setter()".
>> ERROR at line 39: Cannot access non-static function "non_static_func" from a static variable initializer.
